driver/main.cpp

```diff
@@ -52,10 +52,10 @@ int main(int argc, char* argv[])
         std::cout << " " << argv[i];
     std::cout << std::endl;
 
-    Driver* drv = nullptr;
+    std::shared_ptr<Driver> drv;
     for(auto f : rdm::GetRegistry())
     {
-        drv = f(base_arg);
+        drv = std::shared_ptr<Driver>(f(base_arg));
```

Comment:
```suggestion
        drv.reset(f(base_arg));
```
Or at least `make_shared`. But I think `.reset` is the most suitable.

Reply:
You can't actually use make_shared in this case since Driver is pure virtual, but reset makes sense to me.
Ill swap it.